File: DA/Models/DTOs/BackendDetailDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DA.Models.DTOs {
    public class BackendDetailDTO {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public double Price { get; set; }
        public int ZipCode { get; set; }
        public string City { get; set; }
        public string Street { get; set; }
        public int StreetNumber { get; set; }
        public string Country { get; set; }
        public string Organizer { get; set; }
        public string ContactPerson { get; set; }
        public string Email { get; set; }
        public string website { get; set; }
        public string Phone { get; set; }
        public bool IsFree { get; set; }
        public List<ParticipantDTO> participants { get; set; }
        public int availablePlaces { get; set; }
    }
}

File: DA/ClientApp/src/app/backend-detail/BackendDetailDto.ts
import { Member } from "../registration/Member";
import { CompanyMember } from "./CompanyMember";

export class BackendDetailDto {
  id: Number;
  name: string;
  start: Date;
  end: Date;
  price: Number;
  zipCode: Number;
  city: string;
  street: string;
  streetNumber: Number;
  country: string;
  organizer: string;
  contactPerson: string;
  email: string;
  website: string;
  phone: string;
  isFree: boolean;
  participants: CompanyMember[];
  availablePlaces: Number;

  constructor(id: Number,    name: string,    start: Date,    end: Date,    price: Number,    zipCode: Number,    city: string,    street: string,    streetNumber: Number,    country: string,    organizer: string,
    contactPerson: string,
    email: string,
    website: string,
    phone: string,
    isFree: boolean,
    participants: CompanyMember[],
    availablePlaces: Number) {

    this.id =id;
    this.name = name;
    this.start =  start;
    this.end =  end;
    this.price =  price;
    this.zipCode =  zipCode;
    this.city =  city;
    this.street =  street;
    this.streetNumber =  streetNumber;
    this.country =  country;
    this.organizer =  organizer;
    this.contactPerson =  contactPerson;
    this.email =  email;
    this.website =  website;
    this.phone =  phone;
    this.isFree = isFree;
    this.participants =  participants;
    this.availablePlaces =  availablePlaces;

  }
}
 
